CMakeLists.txt

```diff
@@ -153,6 +153,15 @@ install(FILES ${PROJECT_BINARY_DIR}/source/adios2/common/ADIOSConfig.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/adios2/common
 )
 
+#------------------------------------------------------------------------------#
```

Comment:
Move to `DetectOptions.cmake` as that's where most of this type of logic is.  I've been trying, sometimes successfully, sometimes not, to keep the top level `CMakeLists.txt` as a "main" and offload most of the feature logic elsewhere.